api/api/src/main/scala/uk/ac/wellcome/platform/api/Tracing.scala

```diff
@@ -19,20 +30,41 @@ object Tracing {
           "http://localhost:9200"),
         "secret_token" -> config.getOrElse[String]("apm.secret")("")
       ).asJava)
+    actorSystem.dispatchers.registerConfigurator(
+      "tracing-dispatcher",
+      new TraceableDispatcherConfigurator(
+        actorSystem.dispatchers.defaultDispatcherConfig,
+        actorSystem.dispatchers.prerequisites))
+    ec = actorSystem.dispatchers.lookup("tracing-dispatcher")
+  }
+
+  implicit var ec: ExecutionContext =
+    scala.concurrent.ExecutionContext.Implicits.global
```

Comment:
I don't think we use the global execution context everywhere; usually it gets passed in as an implicit alongside `actorSystem`. Is there a reason we're using it here?

Reply:
Yeah, this was a bit of a pain - it's here so the tests work, in reality it gets mutated by `init` to use the execution context from the actor system (ie the default dispatcher). It is true that if you were to forget to call `init`, this would be "wrong", so if you have any thoughts on how to do it better please say!

Reply:
I don't, but please write that in the code 😉